### DISCUSSION TITLE
Sitemap

### DISCUSSION THREAD

**Discussion by danywebxx** [cotes2020/jekyll-theme-chirpy#1815] [Q&A] [answered]
Hello team,

I'm happily using your project for my blog https://danyweb.it But I don't understand how sitemap.xml is generated. 

Can someone explain this to a dumbass like me?

**Comment by leessang01** [comment, marked as answer]
Hi,
AFAIK, [jekyll-sitemap](https://github.com/jekyll/jekyll-sitemap) is used to generate `sitemap.xml` by default. You can see this sentence on their readme.

> Jekyll plugin to silently generate a sitemaps.org compliant sitemap for your Jekyll site

However, I think `sitemap.xml` generated by jekyll-sitemap has a little unnecessary URLs such as tags, archives, about, etc. So you can make your own `sitemap.xml` generated by liquid. You just put `sitemap.xml` file (see below) on your root directory and it will work.

Here is the content of `sitemap.xml` I used. You can see [my sitemap](https://leessang01.github.io/sitemap.xml) generated by this code.
I hope you find my answer useful.

```
---
layout: null
---
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd" xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
  {% for post in site.posts %}
    <url>
      <loc>{{ site.url }}{{ post.url }}</loc>
      {% if post.lastmod == null %}
        <lastmod>{{ post.date | date_to_xmlschema }}</lastmod>
      {% else %}
        <lastmod>{{ post.lastmod | date_to_xmlschema }}</lastmod>
      {% endif %}

      {% if post.sitemap.changefreq == null %}
        <changefreq>weekly</changefreq>
      {% else %}
        <changefreq>{{ post.sitemap.changefreq }}</changefreq>
      {% endif %}

      {% if post.sitemap.priority == null %}
        <priority>0.5</priority>
      {% else %}
        <priority>{{ post.sitemap.priority }}</priority>
      {% endif %}
    </url>
  {% endfor %}
</urlset>
```

**Comment by MhmRhm** [comment]
Hello guys. I have used this theme at https://mhmrhm.github.io/tutorials/ but Google Search Console says it cannot find my sitemap.xml. Can you help?

**Reply by SoumyaK4** [reply]
@MhmRhm manually add [this link](https://mhmrhm.github.io/tutorials/sitemap) to search console

**Reply by MhmRhm** [reply]
I have done that and the link is indexed as you can see in the attached image, but still have the same issue. Google says it can not fetch the sitemap.

![image](https://github.com/user-attachments/assets/b2252cec-9c00-4671-b802-ffb2dbdd4e67)

**Reply by SoumyaK4** [reply]
@danywebxx can you share the screenshot of where it says it cannot add the sitemap please?

and make sure you added it here:
![image](https://github.com/user-attachments/assets/f0705439-cab2-4554-956d-16be067cf951)

**Reply by MhmRhm** [reply]
Dear @SoumyaK4 here is the sitemap screenshot

![Screenshot 2024-12-20 114319](https://github.com/user-attachments/assets/ec3887b7-017f-46b1-9b8f-dded593e8d56)

**Reply by SoumyaK4** [reply]
@MhmRhm can you create 
`assets/robots.txt` and put this there
```
---
permalink: /robots.txt
# The robots rules
---

User-agent: *
Allow: /

Sitemap: {{ '/sitemap.xml' | absolute_url }}
```

**Reply by MhmRhm** [reply]
@SoumyaK4 you can see the robots file [here](https://github.com/MhmRhm/tutorials/blob/main/assets/robots.txt). It is not working . still get same `Couldn't fetch` message.

<img width="928" alt="Screenshot 2024-12-23 at 7 32 43 AM" src="https://github.com/user-attachments/assets/211d35dd-bbf9-4df4-ae4c-96535a239a89" />

**Reply by MhmRhm** [reply]
@SoumyaK4 Updated the robots file.

**Reply by SoumyaK4** [reply]
@danywebxx I just tested with different sites I have, and its reading if no `.xml` in sitemap.
![image](https://github.com/user-attachments/assets/6b5d477a-dc71-43ac-a58f-918387ab1f63)
but in your case I see none works, maybe try copying your sitemap file's content and manually create a `sitemap.xml` at the root(same place where _config.yml lives) of your repo

Hey @kungfux sorry for the mention,
as you see with the screenshot above, google isn't crawling the `.xml` ones from chirpy. but if I open the link it shows the sitemap normally for my [chirpy mods](https://chirpy.soumyak4.in/sitemap.xml) site, but weirdly in localhost only the xml one shows up and the other one is 404.

The bottom one from my just-the-docs jekyll site is being crawled fine, with or without the .xml extension.
Is it some bug with chirpy sitemap gen?

**Reply by MhmRhm** [reply]
Dear @SoumyaK4 in GitHub workflow when the pipeline finishes I can download and extract generated site. I can see the sitemap there.

**Reply by SoumyaK4** [reply]
@danywebxx yes, but I'm assuming something is happening with the dynamic rendering, so you can test once with self created sitemap

**Reply by MhmRhm** [reply]
Manual sitemap also left me with same result. With .xml and without .xml. Google search console was unable to see it.

**Reply by SoumyaK4** [reply]
@MhmRhm honestly I have no idea why,
can you make a backup of your posts, then delete the repo, and start from the scratch with chirpy starter.
I do this whenever I encounter problems I can't fix

**Comment by 0xKrat0s** [comment]
Hi, i have tried both manually generating and plugin generated sitemap, but google search console still didint fetch the maps.

**Comment by maxdorx** [comment]
Hi, what I did to fix this was.

Fix: add a small plugin to filter them out after the sitemap is generated.

Create a file _plugins/filter_sitemap.rb:

```
Jekyll::Hooks.register :site, :post_write do |site|
  sitemap = File.join(site.dest, "sitemap.xml")
  if File.exist?(sitemap)
    content = File.read(sitemap)

    # remove <url> blocks for /tags/ and /categories/
    filtered = content.gsub(
      %r{<url>\s*<loc>https?://[^<]*/(tags|categories)/.*?</url>}m,
      ""
    )

    File.write(sitemap, filtered)
  end
end
```

my sitemap: https://sudoosama.com/sitemap.xml
